// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/db_control/models/ByocSpec.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2025-10
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfByocSpec = instanceOfByocSpec;
exports.ByocSpecFromJSON = ByocSpecFromJSON;
exports.ByocSpecFromJSONTyped = ByocSpecFromJSONTyped;
exports.ByocSpecToJSON = ByocSpecToJSON;
const runtime_1 = require("../runtime");
const MetadataSchema_1 = require("./MetadataSchema");
/**
 * Check if a given object implements the ByocSpec interface.
 */
function instanceOfByocSpec(value) {
    let isInstance = true;
    isInstance = isInstance && "environment" in value;
    return isInstance;
}
function ByocSpecFromJSON(json) {
    return ByocSpecFromJSONTyped(json, false);
}
function ByocSpecFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'environment': json['environment'],
        'schema': !(0, runtime_1.exists)(json, 'schema') ? undefined : (0, MetadataSchema_1.MetadataSchemaFromJSON)(json['schema']),
    };
}
function ByocSpecToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'environment': value.environment,
        'schema': (0, MetadataSchema_1.MetadataSchemaToJSON)(value.schema),
    };
}
